plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -925,89 +1006,97 @@ public ObjectId getDatabaseID( final String name ) throws KettleException {
   private ObjectId getObjectId( final String name, final RepositoryDirectoryInterface dir,
                                 final RepositoryObjectType objectType, boolean includedDeleteFiles ) {
     final String absPath = getPath( name, dir, objectType );
-    RepositoryFile file = pur.getFile( absPath );
-    if ( file != null ) {
-      // file exists
-      return new StringObjectId( file.getId().toString() );
-    } else if ( includedDeleteFiles ) {
-      switch ( objectType ) {
-        case DATABASE: {
-          // file either never existed or has been deleted
-          List<RepositoryFile>
+
+    readWriteLock.writeLock().lock();
```

Comment:
This looks like it should be the readLock, not the writeLock. I don't see any writing here.